diff -Naur vsftpd-2.3.5.orig/sysutil.c vsftpd-2.3.5/sysutil.c
--- vsftpd-2.3.5.orig/sysutil.c	2012-01-16 17:59:45.000000000 -0500
+++ vsftpd-2.3.5/sysutil.c	2012-01-16 17:59:59.000000000 -0500
@@ -375,6 +375,7 @@
 int
 vsf_sysutil_write(const int fd, const void* p_buf, const unsigned int size)
 {
+  int retry = 0;
   while (1)
   {
     int retval = write(fd, p_buf, size);
@@ -384,6 +385,16 @@
     {
       continue;
     }
+    if (retval < 0 && saved_errno == EAGAIN)
+    {
+      if (retry++ < 5) {
+        struct timespec req;
+        req.tv_sec = 0;
+        req. tv_nsec = 10000000;
+        nanosleep(&req, 0);
+        continue;
+      }
+    }
     return retval;
   }
 }
