diff -Naur util-linux-2.20.1.orig/fdisk/fdiskbsdlabel.h util-linux-2.20.1/fdisk/fdiskbsdlabel.h
--- util-linux-2.20.1.orig/fdisk/fdiskbsdlabel.h	2011-12-12 12:28:53.000000000 -0500
+++ util-linux-2.20.1/fdisk/fdiskbsdlabel.h	2011-12-12 12:29:08.000000000 -0500
@@ -49,7 +49,7 @@
 #if defined (i386) || defined (__sparc__) || defined (__arm__) || \
     defined (__mips__) || defined (__s390__) || defined (__sh__) || \
     defined(__x86_64__) || defined (__avr32__) || defined(__cris__) || \
-    defined(__xtensa__)
+    defined(__xtensa__) || defined(__nios2__)
 #define BSD_LABELSECTOR   1
 #define BSD_LABELOFFSET   0
 #elif defined (__alpha__) || defined (__powerpc__) || defined (__ia64__) || defined (__hppa__)
