diff -Naur attr-2.4.43.orig/include/buildmacros attr-2.4.43/include/buildmacros
--- attr-2.4.43.orig/include/buildmacros	2010-02-01 11:46:26.759656408 -0500
+++ attr-2.4.43/include/buildmacros	2010-02-01 11:46:57.783657043 -0500
@@ -72,10 +72,7 @@
 	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).la $(PKG_DEVLIB_DIR); \
 	../$(INSTALL) -m 644 $(LIBNAME).la $(PKG_DEVLIB_DIR)/$(LIBNAME).la ; \
 	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -T so_base $(LIBNAME).la $(PKG_LIB_DIR); \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).a $(PKG_LIB_DIR)/$(LIBNAME).a; \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).la $(PKG_LIB_DIR)/$(LIBNAME).la; \
-	../$(INSTALL) -S $(PKG_LIB_DIR)/$(LIBNAME).so $(PKG_DEVLIB_DIR)/$(LIBNAME).so
+	../$(INSTALL) -T so_base $(LIBNAME).la $(PKG_LIB_DIR)
 else
 INSTALL_LTLIB_DEV = $(INSTALL_LTLIB_STATIC)
 endif
