--- curl-7.21.7.orig/lib/connect.c	2014-02-13 10:14:39.707314304 -0700
+++ curl-7.21.7/lib/connect.c	2014-02-13 10:20:06.102428434 -0700
@@ -350,10 +350,17 @@
     if(done > 0) {
 #ifdef ENABLE_IPV6
       /* ipv6 address */
-      if((af == AF_INET6) &&
-         (Curl_inet_pton(AF_INET6, myhost, &si6->sin6_addr) > 0)) {
-        si6->sin6_family = AF_INET6;
-        si6->sin6_port = htons(port);
+      if(af == AF_INET6) { 
+        char *scope_ptr = strchr(myhost, '%');
+        if(scope_ptr)
+          *(scope_ptr++) = 0;
+
+        if(Curl_inet_pton(AF_INET6, myhost, &si6->sin6_addr) > 0) {
+          si6->sin6_family = AF_INET6;
+          si6->sin6_port = htons(port);
+          if(scope_ptr)
+            si6->sin6_scope_id = atoi(scope_ptr);
+        }
         sizeof_sa = sizeof(struct sockaddr_in6);
       }
       else
