diff -Naur glib-2.30.2.orig/gio/Makefile.am glib-2.30.2/gio/Makefile.am
--- glib-2.30.2.orig/gio/Makefile.am	2011-12-19 10:51:13.000000000 -0500
+++ glib-2.30.2/gio/Makefile.am	2011-12-19 10:52:53.000000000 -0500
@@ -608,7 +608,7 @@
 	libgio-2.0.la					\
 	$(NULL)
 
-glib_compile_schemas_LDADD = $(top_builddir)/glib/libglib-2.0.la
+glib_compile_schemas_LDADD = $(top_builddir)/glib/libglib-2.0.la $(glib_compile_schemas_HOST_STATIC)
 glib_compile_schemas_SOURCES = \
 	gvdb/gvdb-format.h		\
 	gvdb/gvdb-builder.h		\
diff -Naur glib-2.30.2.orig/gio/Makefile.in glib-2.30.2/gio/Makefile.in
--- glib-2.30.2.orig/gio/Makefile.in	2011-12-19 10:51:13.000000000 -0500
+++ glib-2.30.2/gio/Makefile.in	2011-12-19 10:52:53.000000000 -0500
@@ -1167,7 +1167,7 @@
 	libgio-2.0.la					\
 	$(NULL)
 
-glib_compile_schemas_LDADD = $(top_builddir)/glib/libglib-2.0.la
+glib_compile_schemas_LDADD = $(top_builddir)/glib/libglib-2.0.la $(glib_compile_schemas_HOST_STATIC)
 glib_compile_schemas_SOURCES = \
 	gvdb/gvdb-format.h		\
 	gvdb/gvdb-builder.h		\
