diff -Naur zlib-1.2.5.orig/Makefile.in zlib-1.2.5/Makefile.in
--- zlib-1.2.5.orig/Makefile.in	2010-07-21 13:09:58.000000000 -0400
+++ zlib-1.2.5/Makefile.in	2010-07-21 13:10:42.000000000 -0400
@@ -167,8 +167,8 @@
 	-@if [ ! -d $(DESTDIR)$(sharedlibdir) ]; then mkdir -p $(DESTDIR)$(sharedlibdir); fi
 	-@if [ ! -d $(DESTDIR)$(man3dir)      ]; then mkdir -p $(DESTDIR)$(man3dir); fi
 	-@if [ ! -d $(DESTDIR)$(pkgconfigdir) ]; then mkdir -p $(DESTDIR)$(pkgconfigdir); fi
-	cp $(STATICLIB) $(DESTDIR)$(libdir)
-	cp $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)
+	if test "$(STATICLIB)" -a -f $(STATICLIB) ; then cp $(STATICLIB) $(DESTDIR)$(libdir);fi
+	if test "$(SHAREDLIBV)" -a -f $(SHAREDLIBV) ; then cp $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir);fi
 	cd $(DESTDIR)$(libdir); chmod u=rw,go=r $(STATICLIB)
 	-@(cd $(DESTDIR)$(libdir); $(RANLIB) libz.a || true) >/dev/null 2>&1
 	-@cd $(DESTDIR)$(sharedlibdir); if test "$(SHAREDLIBV)" -a -f $(SHAREDLIBV); then \
