--- httpd-2.2.21.orig/modules/dav/fs/repos.c	2012-03-15 00:46:37.188805645 -0600
+++ httpd-2.2.21/modules/dav/fs/repos.c	2012-03-15 08:30:51.155760749 -0600
@@ -1865,7 +1865,7 @@
         if (resource->collection)
             return DAV_PROP_INSERT_NOTDEF;
 
-        (void) sprintf(buf, "%" APR_OFF_T_FMT, resource->info->finfo.size);
+        (void) sprintf(buf, "%ld", resource->info->finfo.size);
         value = buf;
         break;
 
