diff -Naur dbus-1.4.6.orig/configure dbus-1.4.6/configure
--- dbus-1.4.6.orig/configure	2011-02-21 12:36:47.000000000 -0500
+++ dbus-1.4.6/configure	2011-02-21 12:37:21.000000000 -0500
@@ -19608,18 +19608,18 @@
      ;;
   esac
 
-  case " $CFLAGS " in
-  *[\ \	]-fPIE[\ \	]*) ;;
-  *) if test x$dbus_win = xno && cc_supports_flag -fPIE; then
-        PIE_CFLAGS="-fPIE"
-        if ld_supports_flag -z,relro; then
-           PIE_LDFLAGS="-pie -Wl,-z,relro"
-        else
-           PIE_LDFLAGS="-pie"
-        fi
-     fi
-     ;;
-  esac
+#  case " $CFLAGS " in
+#  *[\ \	]-fPIE[\ \	]*) ;;
+#  *) if test x$dbus_win = xno && cc_supports_flag -fPIE; then
+#        PIE_CFLAGS="-fPIE"
+#        if ld_supports_flag -z,relro; then
+#           PIE_LDFLAGS="-pie -Wl,-z,relro"
+#        else
+#           PIE_LDFLAGS="-pie"
+#        fi
+#     fi
+#     ;;
+#  esac
 
   ### Disabled warnings, and compiler flag overrides
 
